addons/web/static/src/js/services/notification_service.js

```diff
@@ -42,20 +42,22 @@ var NotificationService = AbstractService.extend({
      * @param {number} notificationId
      * @param {boolean} [silent=false] if true, the notification does not call
      *   onClose callback
+     * @param {integer} [wait=0] seconds to wait before closing the notification
      */
-    close: function (notificationId, silent) {
+    close: function (notificationId, silent, wait = 0) {
         var notification = this.notifications[notificationId];
         if (!notification) {
             return;
         }
-        notification.close(silent);
+        setTimeout(() => {
+            notification.close(silent);
+        }, wait);
```

Comment:
You can squash the diff in the right commit.

For this part, I don't know if it changes anything but you are changing the behavior when wait = 0. So if you want to be careful, do not use setTimeout if wait = 0 but if you judge it cannot break any current usecase... I trust you :)